lua/weapons/grav_blaster.lua

```diff
@@ -0,0 +1,232 @@
+
+local globalBursts = {}
+
+local function calculateSpread()
+	return VectorRand() * 0.01
+end
+
+local function makeBurst(pos, dir, owner, weapon)
+	return {
+		pos = pos,
+		dir = dir + calculateSpread(),
+		owner = owner,
+		weapon = weapon,
+		timeCreated = CurTime(),
+	}
+end
+
+local function renderBurst(burst)
+	local fx = EffectData()
+	fx:SetOrigin(burst.pos)
+	fx:SetNormal(burst.dir)
+	fx:SetRadius(15)
+
+	util.Effect("RPGShotDown", fx)
+	util.Effect("CommandPointer", fx)
+end
+
+local function applyGravityToEntity(burst, entity, magnitude)
+	if not IsValid(entity) then
+		return
+	end
+
+	local entityPhysics = entity:GetPhysicsObject()
+	if not IsValid(entityPhysics) then
+		return
+	end
+
+	local tossDirection = (entity:GetPos() - burst.pos):GetNormalized()
+	local tossForce = -tossDirection * magnitude * entityPhysics:GetMass()
+	if entity:IsPlayer() then
+		-- Can't use physics methods on a player
+		-- Arbitrary 0.03 because we're setting velocity, not force
+		-- It also acts different with players, toss players away instead of inwards like for props.
+		entity:SetVelocity(-tossForce * 0.03)
+	else
+		-- And some random angular velocity for good measure
+		entityPhysics:AddAngleVelocity(VectorRand() * 50)
+		entityPhysics:ApplyForceCenter(tossForce)
+	end
+end
+
+local function explodeBurst(burst)
+	if not IsValid(burst.weapon) or not IsValid(burst.owner) then
+		return
+	end
+
+	util.BlastDamage(burst.weapon, burst.owner, burst.pos, BURST_EXPLOSION_RADIUS_UNITS, BURST_DAMAGE_HP)
+	local fx = EffectData()
+	fx:SetOrigin(burst.pos)
+
+	util.Effect("Explosion", fx)
+
+	-- Toss all props inwards and players away in the explosion radius
+	local entities = ents.FindInSphere(burst.pos, BURST_EXPLOSION_RADIUS_UNITS)
+
+	for _, entity in ipairs(entities) do
+		applyGravityToEntity(burst, entity, BURST_TOSS_IMPULSE_MAGNITUDE)
+	end
+end
+
+local function updateBurst(burst)
+	local timeElapsed = CurTime() - burst.timeCreated
+	if timeElapsed > BURST_LIFETIME_SECS then
+		return true
+	end
+
+	-- check if the burst has collided
+	local endPos = burst.pos + burst.dir * BURST_SPEED_UNITS * timeElapsed
+	local trace = util.TraceLine({
+		start = burst.pos,
+		endpos = endPos,
+		filter = {
+			burst.owner,
+		},
+	})
+
+	if trace and trace.Hit then
+		return true
+	end
+
+	local entities = ents.FindInSphere(endPos, BURST_PASSIVE_RADIUS_UNITS)
+	for _, entity in ipairs(entities) do
+		if entity == burst.owner then
+			continue
+		end
+
+		applyGravityToEntity(burst, entity, BURST_PASSIVE_IMPULSE_MAGNITUDE)
+	end
+
+	burst.pos = endPos
+
+	return false
+end
+
+function SWEP:FireBurst(position, direction)
+	if CLIENT then
+		return
+	end
+
+	table.insert(globalBursts, makeBurst(position, direction, self:GetOwner(), self))
+end
+
+function SWEP:Initialize()
+	self:SetHoldType("physgun")
+end
+
+function SWEP:Reload()
+	if self:GetOwner():GetAmmoCount(self.Primary.Ammo) == 0 then
```

Comment:
You use `self:GetOwner()` four times in this function, should probably store it in a variable.